service/src/main/java/org/wildfly/service/Installer.java

```diff
@@ -0,0 +1,393 @@
+                return this.builder();
+            }
+
+            @Override
+            public B asActive() {
+                this.mode = ServiceController.Mode.ACTIVE;
+                return this.builder();
+            }
+
+            @Override
+            public B withAlias(ServiceName alias) {
+                this.alias = Optional.of(alias);
+                return this.builder();
+            }
+
+            @Override
+            public B withDependency(Dependency<DSB> dependency) {
+                this.dependency = dependency;
+                return this.builder();
+            }
+
+            @Override
+            public B onStart(Consumer<V> consumer) {
+                this.startTask = consumer;
+                return this.builder();
+            }
+
+            @Override
+            public B onStop(Consumer<T> consumer) {
+                this.stopTask = consumer;
+                return this.builder();
+            }
+
+            @Override
+            public Function<T, V> getMapper() {
+                return this.mapper;
+            }
+
+            @Override
+            public Supplier<T> getFactory() {
+                return this.factory;
+            }
+
+            @Override
+            public ServiceController.Mode getInitialMode() {
+                return this.mode;
+            }
+
+            @Override
+            public Optional<ServiceName> getAlias() {
+                return this.alias;
+            }
+
+            @Override
+            public Dependency<DSB> getDependency() {
+                return this.dependency;
+            }
+
+            @Override
+            public Consumer<V> getStartTask() {
+                return this.startTask;
+            }
+
+            @Override
+            public Consumer<T> getStopTask() {
+                return this.stopTask;
+            }
+        }
+    }
+
+    /**
+     * A generic service implementation that provides a single mapped value from a given factory.
+     * @param <T> the source value type
+     * @param <V> the provided value type of the service
+     */
+    class DefaultService<T, V> implements Service {
+
+        private final Consumer<V> consumer;
+        private final Function<T, V> mapper;
+        private final Supplier<T> factory;
+        private final Consumer<T> stopTask;
+
+        private volatile T value;
+
+        /**
+         * Constructs a generic service that provides a mapped value from the specified factory to the specified consumer on {@link Service#start(StartContext)} and consumed by the specified destroyer on {@link Service#stop(StopContext)}.
+         * @param consumer the consumer that accepts the provided value
+         * @param mapper a function that maps the supplied value
+         * @param factory a supplier of this service's provided value
+         * @param destroyer a consumer that destroys the provided value on {@link Service#stop(StopContext)}.
```

Comment:
This param doc doesn't match the code.

Reply:
Will fix.
FYI - I'm in the process of revising this branch to better leverage the changes introduced by WFCORE-6400.  I'll add the hold label for now.